cli/create.go

```diff
@@ -259,6 +259,16 @@ func createContainer(ociSpec oci.CompatOCISpec, containerID, bundlePath,
 		return vc.Process{}, err
 	}
 
+	// Add the ephemeral device if ephemeral volume
+	// has to be attached to the container
+	for _, mnt := range contConfig.Mounts {
+		if IsEphemeralStroage(mnt.Source) {
+			deviceInfo := config.DeviceInfo{}
+			deviceInfo.DevType = "e"
+			deviceInfo.ContainerPath = mnt.Source
+			contConfig.DeviceInfos = append(contConfig.DeviceInfos, deviceInfo)
```

Comment:
I maybe missed something.

Why do you have to convert ephemeral volume to a fake device, and then convert it back to grpc.Storage in virtcontainers? A DevType of "e" is nonexistent so it should work too to add a fake mount type as "ephemeral" and set the contConfig.Mounts[x].Type to "ephemeral".

And deviceInfo is often used to insert some kinds of devices to hypervisor, so I don't think it's appropriate for carrying the ephemeral volume

Reply:
My initial PR didn't use device. It initiated grpc.Storage directly from virtcontainers library. But @bergwolf suggested [here](https://github.com/kata-containers/runtime/pull/307#pullrequestreview-125180817) that we should move the detection to `cli` because `virtcontainers` wasn't the right place. 

```
Please define the new device type in device/config/config.go, and compare m.Source with devInfo.ContainerPath instead.

And for epheVol, you don't need to put it in the shared mounts. Let me explain how it works:

1.on kata cli side, add a device to container config, where devinfo.type is epheVol, and a Mount to container config, where m.Source == devinfo.ContainerPath
2.in kata runtime, skip epheVol by NOT adding it to sharedMounts
3.in kata_agent.go, create tmpfs storage where its mountpoint is devinfo.ContainerPath
This would follow the same storage/device devinfo semantics as other types of devices/storages.
```

Reply:
Moving detection to `cli` package makes sense, there's no problem for this change.

@bergwolf I don't think we need a special device type, instead a special mount type "ephemeral" should work too in my opinion. WDYT?

Reply:
@WeiZhang555 Yes, it can work too. The difference is that I wanted to stick to a storage model that a storage is a sandbox level resource and containers can share it in its mount slice. It's the same storage model we have in the kata-agent.

Reply:
I see your point, I'm working on device manager refactoring for this device model. But I'm thinking that making the ephemeral volume as a fake device looks weird, because this device has a "fake" device type and can't be hot plugged or appended ...

Reply:
ephemeral volume still can be shared. We do it either by creating a sandbox level storage resource, or by crafting mount.Source to point to the same one. And if we choose to share by letting API callers set the same mount.Source, we'd have to make a static link from mount.Source to `grpc.Storage.Mountpoint` so that kata-agent knows about the sharing as well. The static path conversion and meaningful ephemeral volume mount source are the main cons compared to a real sandbox level storage structure.

In fact, I think devinfo needs to be restructured so that it can describe any device or volume rather than just host local devices. Think of ceph rbd as an example. It is a device that needs hotplug, but there is no host device node for it. <-- I was hoping your device refactoring PR to take care of it but if not, we can do it separately.

Reply:
@WeiZhang555 @bergwolf So I presume we have to wait till @WeiZhang555 submits new PR to refactor device manager and then we will restructure this PR according to new refactored device manager? Just trying to understand what would be the next steps.

Reply:
@harche I think the patch is in reasonable shape now. We can update the ephemeral storage later when the device refactor is done.

Reply:
@bergwolf Thanks.

Reply:
device manager rework PR is almost done: https://github.com/kata-containers/runtime/pull/301
But I met some CI problem(probably related to CI itself not the PR)

Reply:
@WeiZhang555 Should I modify this PR to work with the changes in  #301?

Reply:
CI has some problem, so I'm not 100% sure about correctness of #301 , so maybe you can wait.

Or you can go on with your work, I'll rebase on your changes, it's fine for me.